site/docs/plugins/commands.md

```diff
@@ -5,10 +5,682 @@ next: false
+filters.
+
+Let's use the previous knowledge to inspect the next example:
+
+```js
+const myCommands = new CommandGroup();
+myCommands.command("dad", "calls dad", () => {}, { prefix: "?" })
+  .localize("es", "papa", "llama a papa")
+  .localize("fr", "pere", "appelle papa");
+
+const otherCommands = new CommandGroup();
+otherCommands.command("bread", "eat a toast", () => {})
+  .localize("es", "pan", "come un pan")
+  .localize("fr", "pain", "manger du pain");
+
+// Register Each
+
+// Let's assume the user is french and typed /Papi
+bot
+  // this filter will trigger for any command-like as '/regular' or '?custom'
+  .filter(commandNotFound([myCommands, otherCommands], {
+    ignoreLocalization: true,
+    ignoreCase: true,
+  }))
+  .use(async (ctx) => {
+    ctx.commandSuggestion === "?papa"; // evaluates to true
+
+    /* if the ignoreLocalization was falsy instead
+     * we would have gotten:
+     * ctx.commandSuggestion equals "/pain"
+     */
+  });
+
+/* We could add more filters like the above,
+ * with different parameters or CommandGroups to check against
+ */
+```
+
+There is a lot of possibilities!
+
+## Command Options
+
+There are a few options that can be specified per command, per scope, or
+globally for a `CommandGroup` instance. These options allow you to further
+customize how your bot handles commands, giving you more flexibility.
+
+### ignoreCase
+
+By default commands will match the user input in a case-sensitive manner. Having
+this flag set, for example, in a command named `/dandy` will match `/DANDY` the
+same as `/dandY` or any other case-only variation.
+
+### targetedCommands
+
+When users invoke a command, they can optionally tag your bot, like so:
+`/command@bot_username`. You can decide what to do with these commands by using
+the `targetedCommands` config option. With it you can choose between three
+different behaviors:
+
+- `ignored`: Ignores commands that mention your bot's user
+- `optional`: Handles both commands that do and that don't mention the bot's
+  user
+- `required`: Only handles commands that mention the bot's user
+
+### prefix
+
+Currently, only commands starting with `/` are recognized by Telegram and, thus,
+by the [command handling done by the grammY core library](../guide/commands). In
+some occasions, you might want to change that and use a custom prefix for your
+bot. That is made possible by the `prefix` option, which will tell the Commands
+plugin to look for that prefix when trying to identify a command.
+
+If you ever need to retrieve `botCommand` entities from an update and need them
+to be hydrated with the custom prefix you have registered, there is a method
+specifically tailored for that, called `ctx.getCommandEntities(yourCommands)`.
+It returns the same interface as `ctx.entities('bot_command')`
+
+:::tip
+Commands with custom prefixes cannot be shown in the Commands Menu.
+:::
+
+### matchOnlyAtStart
```

Comment:
I'm not sure either if we'd like to refactor now, because this would be breaking
the rationale behind naming it this way is because this is the default behavior of the core lib, and (at least when I started it) I wanted general behavior to match the default lib, with features (like matching in the middle of the text) being opt-in

Reply:
Alright